### ISSUE TITLE
Legend color is not in sync with plot color on change

### ISSUE THREAD

**Issue by shefalijoshi** [nasa/openmct#5233]
#### Summary
Legend needs to update swatch color when user sets color from Plot Series area in Inspector.

#### Expected vs Current Behavior
User currently must nav away and back to see the change, which is undesirable.

#### Steps to Reproduce
1. Create an overlay plot
2. Add a telemetry endpoint to it
3. Change the color of the series from inspector properties
4. Observe that the plot series color changes, but the legend shows the old color in the swatch.

#### Environment
* Open MCT Version: 2.0.3
* Deployment Type: all
* OS: Mac / Windows
* Browser: All

#### Impact Check List

- [x] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [ ] Does this impact a critical component?
- [x] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?

#### Additional Information
In the image below, the swatch colors don't match the plot colors.

<img width="1167" alt="image" src="https://user-images.githubusercontent.com/5247066/169571937-a2cc4c67-e6b2-44ed-b998-4e72f6ee2741.png">

**Comment by khalidadil** [comment]
Verified Fixed in Testathon on 09/30/22

**Comment by ozyx** [comment]
Not fixed. Testathon 09/30/22

- While still in edit mode after creating overlay plot and dragging on a telemetry object, changed plot series color and it did not update the color in the legend.
- However, clicking "Save" and going back into edit mode *did* update the swatch.

**Comment by khalidadil** [comment]
@ozyx I seem to be able to change colors on telemetry immediately after dragging it into a overlay plot

**Comment by ozyx** [comment]
> @ozyx I seem to be able to change colors on telemetry immediately after dragging it into a overlay plot

The color of the plot did change color, but the color of the swatch did not until I saved / hit edit again. Not sure if this was intended to be fixed in this PR, but it feels weird with my end-user hat on.

**Comment by khalidadil** [comment]
@ozyx ah, I get what you're saying! I see that too. That's a good observation

**Comment by unlikelyzero** [comment]
Build 6 Blocker which was left over

**Comment by charlesh88** [comment]
Verified NOT fixed Testathon 2023-01-24 - and now the color cannot be selected and applied. Will file as a separate issue for this.

**Comment by michaelrogers** [comment]
When testing during testathon 2023-01-24 I was able to successfully change the chart color but the legend was not in sync. So issue does not appear to be fixed.

**Comment by jvigliotta** [comment]
What they said...

**Comment by charlesh88** [comment]
Verified fixed Testathon 2023-01-31. Nice!
